extensions/amp-app-banner/0.1/amp-app-banner.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Layout} from '../../../src/layout';
+import {setStyles} from '../../../src/style';
+import {user} from '../../../src/log';
+import {platform} from '../../../src/platform';
+import {viewerFor} from '../../../src/viewer';
+import {CSS} from '../../../build/amp-app-banner-0.1.css';
+import {urlReplacementsFor} from '../../../src/url-replacements';
+import {xhrFor} from '../../../src/xhr';
+
+const TAG = 'amp-app-banner';
+
+class AmpAppBanner extends AMP.BaseElement {
+
+  /** @override */
+  preconnectCallback(onLayout) {
+    if (platform.isIos()) {
+      this.preconnect.url('https://itunes.apple.com', onLayout);
+    } else if (platform.isAndroid()) {
+      this.preconnect.url('https://play.google.com', onLayout);
+    }
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.CONTAINER;
+  }
+
+  /** @override */
+  buildCallback() {
+    if (!platform.isIos() && !platform.isAndroid()) {
```

Comment:
As in fallback to whatever browser builtin functionality?

Problem is the only cases where this work currently is on Chrome Android and Safari iOS. If the user is on Chrome iOS there's no native support for app store banner for example. Same thing for other browsers on Android.